FWCore/Services/src/InitRootHandlers.cc

```diff
@@ -337,5 +414,17 @@ namespace edm {
       s_ignoreWarnings = true;
     }
 
+    void
+    InitRootHandlers::cachePidInfo()
+    {
+      assert(snprintf(pidString_, pidStringLength_-1, "gdb -quiet -p %d 2>&1 <<EOF |\n"
```

Comment:
Don't do real work inside an assert since a compile switch can make the assert go away.